internal/controller/phases_test.go

```diff
@@ -641,6 +641,10 @@ func TestRepositoryFactory(t *testing.T) {
 			name:     "gitlab repo",
 			fetchURL: "https://gitlab.example.org/api/v4/projects/group%2Fproject/packages/generic/cluster-api-proviver-aws/v1.4.1/path",
 		},
+		{
+			name:     "gitlab hyphenated repo",
+			fetchURL: "https://gitlab-test.example.org/api/v4/projects/group%2Fproject/packages/generic/cluster-api-proviver-aws/v1.4.1/path",
```

Comment:
This is a pretty good idea, but it is a special case.
For example, our private GitLab instance called git.<our domain> can't be changed and no `gitlab` prefix or postfix in name.
It would be nice to have some sort of option which can say that it's a GitLab instance if there is not possible to define it form the FQDN.

But seems to community should define is if suitable approach or not.

Reply:
generally agree - the change to the code to support what i consider 'the more correct approach' wasn't within my allotted time as  it seems like it should live in the [providerConfig](https://github.com/kubernetes-sigs/cluster-api-operator/blob/tsmini.support-any-gitlab/internal/controller/manifests_downloader.go#L82-L83) of `Provider type` - which would be large change - perhaps percolating to all providers.

while this does highlight a special use case - i'm wondering if its _generic enough_

Reply:
We can find a more flexible approach in a follow-up PR

Reply:
Ok, make sense.